src/containerapp/azext_containerapp/_utils.py

```diff
@@ -788,6 +788,35 @@ def _remove_readonly_attributes(containerapp_def):
             del containerapp_def['properties'][unneeded_property]
 
 
+def clean_null_values(d):
+    if isinstance(d, dict):
+        return {
+            k: v
+            for k, v in ((k, clean_null_values(v)) for k, v in d.items())
+            if v
+        }
+    if isinstance(d, list):
+        return [v for v in map(clean_null_values, d) if v]
+    return d
+
+
+def _populate_secret_values(containerapp_def, secret_values):
```

Comment:
@haroonf , don't need to change this logic for now, since we need to release the fix.
Can you help me understand the behavior in CLI with YAML here , when CLI parses the YAML with secretName & no value, does the payload to the API translate this object in JSON payload  just name property & no value i.e{ name: xyz} or. {name: xyz, value: null}. if it is the former is the PATCH API, setting the value to null or is it cleaning these up because CLI explicitly passes null?

Reply:
The deserialization process leaves the json payload to {name: xyz, value: null}. The API returns an error "cannot set secret values to null."